doc: add more to subsystems (#39971)
nodejs/node: doc/guides/contributing/pull-requests.md

### Appendix: subsystems

* `lib/*.js` (`assert`, `buffer`, etc.)
* `benchmark`
* `deps`
* `build`
* `doc`
* `lib / src`
* `test`
* `tools`
* `typings`

Here is the full list of

> [subsystems](https://github.com/nodejs/core-validate-commit/blob/main/lib/rules/subsystem.js).

Reviewer: Consider using `[subsystems][]` here to fit it into a line, and adding that link below.

Author: > Consider using `[subsystems][]` here to fit it into a line, and adding that link below.

Thank you for your advice.

* `test`
* `tools`

> Here is the full list of [subsystems][subsystems].

Reviewer 2: ```suggestion
You can find the full list of supported subsystems in the
[nodejs/core-validate-commit][] repo.
```

* `test`
* `tools`

You can find the full list of supported subsystems in the

> [nodejs/core-validate-commit][] repo.

Reviewer 3: ```suggestion
[nodejs/core-validate-commit][] repository.
```

See b9866098a7f8588970bb9107eadd6deb853c1fa2, b11d3f15b31e7a39d3f6a6e0d3f59f021144e813, 4de6f207dca12164b9403e4d539b843d4d21f820, and bbdc15bbdbb61f4e7d6f388b93ea32175a305bd3.

[https://ci.nodejs.org/]: https://ci.nodejs.org/
[pull request template]: https://raw.githubusercontent.com/nodejs/node/HEAD/.github/PULL_REQUEST_TEMPLATE.md
[running tests]: ../../../BUILDING.md#running-tests

> [subsystems]: https://github.com/nodejs/core-validate-commit/blob/main/lib/rules/subsystem.js

Reviewer 2: ```suggestion
[nodejs/core-validate-commit]: https://github.com/nodejs/core-validate-commit/blob/main/lib/rules/subsystem.js
```
